Fix javadoc warnings

When running javadoc, following warning messages are shown:

```
[INFO] --- maven-javadoc-plugin:2.10.1:jar (attach-javadocs) @ spring-cloud-dataflow-docs ---
Downloading: http://repo.spring.io/libs-snapshot-local/woodstox/wstx-asl/3.2.2/wstx-asl-3.2.2.pom
[INFO]
6 warnings
[WARNING] Javadoc Warnings
[WARNING] javadoc: warning - Malformed URL: http://docs.spring.io/autorepo/docs/spring-security/${spring-security.version}/apidocs/
[WARNING] /Users/igopinatha/workspace/git/ilayaperumalg/spring-cloud-dataflow/spring-cloud-dataflow-docs/target/distro-javadoc-sources/spring-cloud-dataflow-core-1.0.0.BUILD-SNAPSHOT-sources/org/springframework/cloud/dataflow/core/StreamAppDefinition.java:179: warning - Tag @see: can't find label in org.springframework.cloud.dataflow.core.DataFlowAppDefinition
[WARNING] /Users/igopinatha/workspace/git/ilayaperumalg/spring-cloud-dataflow/spring-cloud-dataflow-docs/target/distro-javadoc-sources/spring-cloud-dataflow-core-1.0.0.BUILD-SNAPSHOT-sources/org/springframework/cloud/dataflow/core/StreamAppDefinition.java:166: warning - @param argument "name" is not a parameter name.
[WARNING] /Users/igopinatha/workspace/git/ilayaperumalg/spring-cloud-dataflow/spring-cloud-dataflow-docs/target/distro-javadoc-sources/spring-cloud-dataflow-core-1.0.0.BUILD-SNAPSHOT-sources/org/springframework/cloud/dataflow/core/TaskDefinition.java:151: warning - Tag @see: can't find label in org.springframework.cloud.dataflow.core.DataFlowAppDefinition
[WARNING] /Users/igopinatha/workspace/git/ilayaperumalg/spring-cloud-dataflow/spring-cloud-dataflow-docs/target/distro-javadoc-sources/spring-cloud-dataflow-core-1.0.0.BUILD-SNAPSHOT-sources/org/springframework/cloud/dataflow/core/TaskDefinition.java:138: warning - @param argument "name" is not a parameter name.
[WARNING] /Users/igopinatha/workspace/git/ilayaperumalg/spring-cloud-dataflow/spring-cloud-dataflow-docs/target/distro-javadoc-sources/spring-cloud-dataflow-shell-1.0.0.BUILD-SNAPSHOT-sources/org/springframework/cloud/dataflow/shell/ShellCommandLineParser.java:54: warning - @lnk is an unknown tag.
[INFO] Building jar: /Users/igopinatha/workspace/git/ilayaperumalg/spring-cloud-dataflow/spring-cloud-dataflow-docs/target/spring-cloud-dataflow-docs-1.0.0.BUILD-SNAPSHOT-javadoc.jar
[INFO] NOT adding javadoc to attached artifacts list.
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ spring-cloud-dataflow-docs ---
[WARNING] JAR will be empty - no content was marked for inclusion!
```

--- a/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamAppDefinition.java
+++ b/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamAppDefinition.java
@@ -158,7 +158,7 @@ public class StreamAppDefinition extends DataFlowAppDefinition {
 		/**
 		 * Set the name of the app in the registry.
 		 *
-		 * @param name name of app in registry
+		 * @param registeredAppName name of app in registry
 		 * @return this builder object
 		 *
 		 * @see DataFlowAppDefinition#registeredAppName
@@ -173,8 +173,6 @@ public class StreamAppDefinition extends DataFlowAppDefinition {
 		 *
 		 * @param label name of app label
 		 * @return this builder object
-		 *
-		 * @see DataFlowAppDefinition#label
 		 */
 		public Builder setLabel(String label) {
 			this.label = label;

--- a/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinition.java
+++ b/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinition.java
@@ -130,7 +130,7 @@ public class TaskDefinition extends DataFlowAppDefinition {
 		/**
 		 * Set the name of the app in the registry.
 		 *
-		 * @param name name of app in registry
+		 * @param registeredAppName name of app in registry
 		 * @return this builder object
 		 *
 		 * @see DataFlowAppDefinition#registeredAppName
@@ -145,8 +145,6 @@ public class TaskDefinition extends DataFlowAppDefinition {
 		 *
 		 * @param label name of app label
 		 * @return this builder object
-		 *
-		 * @see DataFlowAppDefinition#label
 		 */
 		public TaskDefinitionBuilder setLabel(String label) {
 			this.label = label;

--- a/spring-cloud-dataflow-docs/pom.xml
+++ b/spring-cloud-dataflow-docs/pom.xml
@@ -79,8 +79,7 @@
 									<quiet>true</quiet>
 									<stylesheetfile>${basedir}/src/main/javadoc/spring-javadoc.css</stylesheetfile>
 									<links>
-										<link>http://docs.spring.io/spring-framework/docs/${spring.version}/javadoc-api/</link>
-										<link>http://docs.spring.io/autorepo/docs/spring-security/${spring-security.version}/apidocs/</link>
+										<link>http://docs.spring.io/spring-framework/docs/${spring-framework.version}/javadoc-api/</link>
 										<link>http://docs.spring.io/spring-shell/docs/current/api/</link>
 									</links>
 								</configuration>

--- a/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/apps/package-info.java
+++ b/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/apps/package-info.java
@@ -1,0 +1,4 @@
+/**
+ * Package for the classes that have common application configuration properties.
+ */
+package org.springframework.cloud.dataflow.server.config.apps;

--- a/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/support/package-info.java
+++ b/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/support/package-info.java
@@ -1,5 +1,5 @@
 /**
- * Provides RDMS-related support classes including
+ * Provides RDBMS-related support classes including
  * {@link org.springframework.cloud.dataflow.server.repository.support.PagingQueryProvider}
  * implementations for supported databases.
  */

--- a/spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/ShellCommandLineParser.java
+++ b/spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/ShellCommandLineParser.java
@@ -49,7 +49,7 @@ public class ShellCommandLineParser {
      * Parse {@link ShellProperties} and {@link ApplicationArguments} to create an instance of the
      * @param shellProperties the shell properties
      * @param applicationArguments the raw unprocessed arguments that were passed to the application.
-     * @return a new {@lnk CommandLine} instance.
+     * @return a new {@link CommandLine} instance.
      */
     public CommandLine parse(ShellProperties shellProperties, String[] applicationArguments) {
         List<String> commands = new ArrayList<String>();

